Added new recombination maps for HapMap and deCode lifted over to GRCh38

saurabhbelsare opened the pull request:
This is to complete https://github.com/popsim-consortium/stdpopsim/issues/543

The lifted over genetic maps have been added to AWS and the corresponding details have been updated in HomSap/genetic_maps.py

New map options HapMapII_GRCh38 and DeCodeSexAveraged_GRCh38 have been created in addition to the original HapMapII_GRCh37 and DeCodeSexAveraged_GRCh36.

petrelharp commented:
Hm - something's going wrong with the tests. The files seem okay, so I'm restarting the tests?

Also - this is just adding the new maps. Shouldn't we remove the older ones?

saurabhbelsare commented:
I think I've figured out where this error is coming from. The lifted over maps are not directly inside the tarball that is linked but instead inside a directory inside the tarball. So, after downloading, the maps which should be located in `test_cache/genetic_maps/HomSap/DeCodeSexAveraged_GRCh38/` are instead located in `test_cache/genetic_maps/HomSap/DeCodeSexAveraged_GRCh38/DeCodeSexAveraged_GRCh36_liftedOverTo_Hg38/`. I have created new tarballs which do not have these extra directories. @andrewkern Can you change the uploaded tarballs to these new files: [DeCodeSexAveraged_GRCh38.tar.gz](https://drive.google.com/file/d/1bSlSWj-OpdStfxp8zFH8Kw5IZIEsX1yI/view?usp=sharing)  and [HapMapII_GRCh38.tar.gz](https://drive.google.com/file/d/1bJRB81kQ7ANVwpi5wc1hilNYZcgo7qo5/view?usp=sharing)? I think by changing to these new tarballs, the error should go away. Sorry about this. 

Regarding having the old maps @petrelharp, I spoke with @andrewkern and we discussed having both still in there, in case someone wanted to access the old ones? But I can remove them if that would be preferable.

andrewkern commented:
okay these files are now uploaded to AWS.

andrewkern commented:
i'm going to try to re-trigger the tests

andrewkern commented:
@saurabhbelsare these are now failing because the URLs need to be updated. Links to the new files are below. Please be sure to update the SHA256 hashes as well

https://stdpopsim.s3.us-west-2.amazonaws.com/genetic_maps/HomSap/DeCodeSexAveraged_GRCh38.tar.gz

https://stdpopsim.s3.us-west-2.amazonaws.com/genetic_maps/HomSap/HapMapII_GRCh38.tar.gz

saurabhbelsare commented:
I've updated the links and the sha256 hashes, squashed the commits, and pushed again. While running tests locally, I had to run them twice to get them all to pass, but then they did. Hopefully this should not run into any problems now.

andrewkern commented:
looks good @saurabhbelsare. now passing all checks.

one question here-- do we need more documentation on how the liftover was done?

saurabhbelsare commented:
The liftover was done using the script created by Jeff Adrion as outlined here: https://github.com/popsim-consortium/stdpopsim/issues/571#issuecomment-692374581 Here are the exact commands I used:

`python <path_to_stdpopsim>/stdpopsim/maintenance/liftOver_catalog.py --species HomSap --map HapMapII_GRCh37 --chainFile <path_to_chainfiles>/chainfiles/hg19ToHg38.over.chain.gz --validationChain ../../chainfiles/hg38ToHg19.over.chain.gz --winLen 1000 --useAdjacentAvg --retainIntermediates --gapThresh 1000000`

`python <path_to_stdpopsim>/stdpopsim/maintenance/liftOver_catalog.py --species HomSap --map DeCodeSexAveraged_GRCh36 --chainFile <path_to_chainfiles>/chainfiles/hg18ToHg38.over.chain.gz --winLen 1000 --useAdjacentAvg --retainIntermediates --gapThresh 1000000`

The only difference is that the validation chain file does not exist for hg36tohg38, so the validation was performed manually by plotting the rates for one assembly vs another for each chromosome.

Where should these be included as documentation, if we want to do that?

andrewkern commented:
would be great if this could be added as documentation-- yes.

andrewkern commented:
maybe just putting that in the `long_description` slot would be okay for now?

andrewkern commented on stdpopsim/catalog/HomSap/genetic_maps.py:
```diff
@@ -34,6 +34,35 @@
+        map previously lifted over to GRCh37. 
+        Liftover was performed using the liftOver_catalog.py script from 
+        stdpopsim/maintainance. Exact command used is as follows:
+        `python <path_to_stdpopsim>/stdpopsim/maintenance/liftOver_catalog.py --species HomSap 
```
```suggestion


        `python <path_to_stdpopsim>/stdpopsim/maintenance/liftOver_catalog.py --species HomSap 
```
```suggestion
        `python <path_to_stdpopsim>/stdpopsim/maintenance/liftOver_catalog.py --species HomSap 
```

andrewkern replied:
maybe add white space here?

andrewkern commented on stdpopsim/catalog/HomSap/genetic_maps.py:
```diff
@@ -63,6 +92,48 @@
+        GRCh38 using liftover.
+        Liftover was performed using the liftOver_catalog.py script from 
+        stdpopsim/maintainance. Exact command used is as follows:
+        `python <path_to_stdpopsim>/stdpopsim/maintenance/liftOver_catalog.py --species HomSap 
```
```suggestion


        `python <path_to_stdpopsim>/stdpopsim/maintenance/liftOver_catalog.py --species HomSap 
```
```suggestion
        `python <path_to_stdpopsim>/stdpopsim/maintenance/liftOver_catalog.py --species HomSap 
```

andrewkern replied:
same as above-- maybe add whitespace?

andrewkern commented:
looks like pre-commit checks are failing now.. some sort of linting issue.

saurabhbelsare commented:
Pushed again after running pre-commit first.

andrewkern commented:
okay looks like we are good here
